wordpress-mobile/WordPress-iOS
# Support publicize on posting

Fixes #2698 

To test:

1 - Configure at least one Publicize Connection for a Blog
2 - Create a new Post
3 - Navigate to the Post Settings
4 - Edit the Publicize Message
5 - Enable/Disable some connections for the Post
6 - Publish the Post and check that it's shared to the selected Publicize Connections with the correct message

When sharing is not supported or there are no Connections setup the Sharing section is completely hidden.
If the Post has been published the Sharing section and its items are disabled.

Needs review:  @jleandroperez @kurzee @koke 
CC: @sendhil 

### **Some background to understand the implementation**
How the API handles enabling/disabling Publicize Connections:

**_Fetching disabled connections_**
Disabled publicize connections are returned for every post in the metadata section:

```
...
"metadata": [ 
    {
        "id": "270",
        "key": "_wpas_skip_16508268",
        "value": "1"
    }
],
....
```

The previous excerpt shows that the connection with connectinoKeyringID = 16508268 is disabled for that post.

**_Updating disabled connections_**

This is done in the same way as with any other metadata entry.

POST data to *disable a connection*

```
...
"metadata": [ 
    {
        "key": "_wpas_skip_16508268",
        "value": "1",
        "operation": "add"
    }
],
....
```

POST data to *enable a connection*

```
...
"metadata": [ 
    {
        "id": "270",
        "key": "_wpas_skip_16508268",
        "value": "0",
        "operation": "update"
    }
],
....
```

Please note that in order to re-enable a connection it is necessary to issue an update operation with the previously retrieved id and "0" as the value. This is why it is not enough to just store the disabled connection keyringIDs.
Calypso currently issues a delete operation to accomplish this and it does not work (https://github.com/Automattic/wp-calypso/issues/11099).
